### DISCUSSION TITLE
6.2 Step 7, will not save in Geoserver (again)

### DISCUSSION THREAD

**Discussion by jenw-ua** [ua-gist604b-f24/syllabus#39] [General]
Hello all, I really appreciate the help everyone has been offering...thanks so much.

The newest, yet continuous problem I'm having is in Geoserver. I'm on step 7 of module 6.2 again (I've deleted and redone it before and now again, hoping something might have improved).  It literally just will NOT save! It's like the button is dead. It will not save the file. That's it.

I've retried this all today (did it multiple times last week as well), I've used Safari, Chrome, and Firefox. I've killed and redone my codespace multiple times. I'm using 8-core. I've talked nicely to it...

This is the error I'm getting when I look with the developer tools...when I'm not using developer tools, you can't tell anything is even happening. It's like the "save" button is broke or unplugged. So I'm stuck at step 7. 

Please help or send good vibes into the atmosphere. Hope it's working for everyone else! And Happy Thanksgiving week to you all!

<img width="1468" alt="Screenshot 2024-11-25 at 10 02 23 AM" src="https://github.com/user-attachments/assets/26acd820-3d43-4a6c-be78-b07cfd4b3570">

**Comment by aaryno** [comment]
The 403 Forbidden error is possibly a CORS error. This has plagued us in the past. It's a security thing where a webpage on one site can't include dynamic content from another site. The way geoserver constructs the links doesn't mesh well with how the browser understands urls and so some of the content is blocked. It's not an easy problem to solve. It only is blocked on the web interface, however. There is a REST url that will allow you to bypass that. https://docs.geoserver.org/latest/en/user/rest/styles.html.

Save your SLD file to your codespace. For this example, call it "my_style.sld"

From your terminal:

curl -v -u admin:geoserver -XPOST -H "Content-type: text/xml" -d @my_style.sld http://localhost:8080/geoserver/rest/styles

**Reply by jenw-ua** [reply]
I guess I'm a little confused. I saw this info about the 403 Forbidden error, but in 6.2 Step 7 aren't we trying to save the update to the osm store by updating the template? It's the point at 17:20 in the 7.2 video. I'm kind of confused about what sld file we are talking about in 6.2 step 7 and how that applies to updating the store? It will not save the update I'm trying to make to the store. Just trying to understand...
